bin/update-schema.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Update the bundled schema version tests run against.
+#
+# The schema is pulled from mozilla-pipeline-schemas at the following URL:
+#
+#     https://raw.githubusercontent.com/mozilla-services/mozilla-pipeline-schemas/$HASH/schemas/glean/glean/glean.1.schema.json
+#
+# References in the code base are updated.
+#
+# Usage: update-schema.sh <commit hash>
+#
+# Environment:
+#
+# DRY_RUN - Do not modify files or run destructive commands when set.
+# VERB    - Log commands that are run when set.
+
+set -eo pipefail
+
+run() {
+  [ "${VERB:-0}" != 0 ] && echo "+ $*"
+  if [ "$DOIT" = y ]; then
+      "$@"
+  else
+      true
+  fi
+}
+
+update() {
+  COMMIT_HASH="$1"
+  FULL_URL="$(printf "$SCHEMA_URL" "$COMMIT_HASH")"
+  SCHEMA_PATH="${WORKSPACE_ROOT}/glean-core/preview/tests/glean.1.schema.json"
+
+  echo "Vendoring schema from ${FULL_URL}"
+  run curl --silent --fail --show-error --location --retry 5 --retry-delay 10 "$FULL_URL" --output "$SCHEMA_PATH"
+
+  # Update the android/build.gradle version
+
+  FILE=glean-core/android/build.gradle
+  run $SED -i.bak -E \
+      -e "s/^String GLEAN_PING_SCHEMA_GIT_HASH = \"[0-9a-z.-]+\"/String GLEAN_PING_SCHEMA_GIT_HASH = \"${COMMIT_HASH}\"/" \
+      "${WORKSPACE_ROOT}/${FILE}"
+  run rm "${WORKSPACE_ROOT}/${FILE}.bak"
```

Comment:
Either here or as a follow-on -- it might be nice to use a vendored file for this too (rather than a git hash).  Also for the Python checking below.  That would save the network request / caching etc.

Reply:
True, but I don't think the glean_parser supports that currently. I'll open a bug for that.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1618263